crates/cli-support/src/js/binding.rs

```diff
@@ -905,6 +905,19 @@ fn instruction(
             js.push(format!("isLikeNone({0}) ? {1} : {0}", val, hole));
         }
 
+        Instruction::F64FromOptionSentinelInt => {
+            let val = js.pop();
+            js.cx.expose_is_like_none();
+            js.assert_optional_number(&val);
+            js.push(format!("isLikeNone({0}) ? 0x100000001 : {0}", val));
+        }
+        Instruction::F64FromOptionSentinelF32 => {
+            let val = js.pop();
+            js.cx.expose_is_like_none();
+            js.assert_optional_number(&val);
+            js.push(format!("isLikeNone({0}) ? 0x100000001 : ({0} === 0x100000001 ? 0x100000002 : {0})", val));
```

Comment:
```suggestion
            js.push(format!("isLikeNone({0}) ? 0x100000001 : Math.fround({0})", val));
```
When you pass a `Number` into a `f32` parameter in WebAssembly, [it gets rounded to the nearest representable value](https://www.w3.org/TR/2024/WD-wasm-js-api-2-20241007/#:~:text=Let%20f32%20be%20number%20rounded%20to%20the%20nearest%20representable%20value%20using%20IEEE%20754%2D2008%20round%20to%20nearest%2C%20ties%20to%20even%20mode.%20%5BIEEE%2D754%5D).
We can apply the same behavior by using [`Math.fround()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/fround), which uses the [same method from the same spec](https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-math.fround), albeit a newer version of it (2008 vs 2019), which [according to Wikipedia](https://en.wikipedia.org/wiki/IEEE_754#History) shouldn't affect us.

I assume this is still worth it compared to accessing the array buffer.
If not, we can always exempt `f32` from this improvement.

Reply:
Thank you for looking into this! I hadn't even thought about `fround`. It's perfect!

Alright, with this, I have all the puzzle pieces together to handle all edge cases.

> which uses the [same method from the same spec](https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-math.fround), albeit a newer version of it (2008 vs 2019), which [according to Wikipedia](https://en.wikipedia.org/wiki/IEEE_754#History) shouldn't affect us.

I can just add a test to make sure it doesn't affect us.